packages/grafana-ui/src/components/FileUpload/FileUpload.tsx

```diff
@@ -40,13 +40,16 @@ export const FileUpload: FC<Props> = ({
   const style = getStyles(theme, size);
   const [fileName, setFileName] = useState('');
 
-  const onChange = useCallback((event: FormEvent<HTMLInputElement>) => {
-    const file = event.currentTarget?.files?.[0];
-    if (file) {
-      setFileName(file.name ?? '');
-    }
-    onFileUpload(event);
-  }, []);
+  const onChange = useCallback(
+    (event: FormEvent<HTMLInputElement>) => {
+      const file = event.currentTarget?.files?.[0];
+      if (file) {
+        setFileName(file.name ?? '');
+      }
+      onFileUpload(event);
+    },
+    [onFileUpload]
```

Comment:
nit: shouldn't setFileName be part of dependencies?

Reply:
nope the useState's setter function will not change https://stackoverflow.com/questions/59709304/setstate-in-reacts-useeffect-dependecy-array